gpt_oss/metal/source/sdpa.metal

```diff
@@ -142,23 +151,34 @@ kernel void gptoss_f32_sdpa_q8_d64(
         m6 = new_m6;
         m7 = new_m7;
 
-        const float2 vval = reinterpret_cast<const device float2*>(v)[tid];
-        v += 2 * num_kv_heads * head_dim;
-        out0 = metal::fma(vval, qk0, out0 * alpha0);
-        out1 = metal::fma(vval, qk1, out1 * alpha1);
-        out2 = metal::fma(vval, qk2, out2 * alpha2);
-        out3 = metal::fma(vval, qk3, out3 * alpha3);
-        out4 = metal::fma(vval, qk4, out4 * alpha4);
-        out5 = metal::fma(vval, qk5, out5 * alpha5);
-        out6 = metal::fma(vval, qk6, out6 * alpha6);
-        out7 = metal::fma(vval, qk7, out7 * alpha7);
+        // Optimized attention-weighted value accumulation
+        out0 = metal::fma(vval, exp0, out0);
+        out1 = metal::fma(vval, exp1, out1);
+        out2 = metal::fma(vval, exp2, out2);
+        out3 = metal::fma(vval, exp3, out3);
+        out4 = metal::fma(vval, exp4, out4);
+        out5 = metal::fma(vval, exp5, out5);
+        out6 = metal::fma(vval, exp6, out6);
+        out7 = metal::fma(vval, exp7, out7);
     }
-    reinterpret_cast<device float2*>(output + 0 * head_dim)[tid] = out0 / l0;
-    reinterpret_cast<device float2*>(output + 1 * head_dim)[tid] = out1 / l1;
-    reinterpret_cast<device float2*>(output + 2 * head_dim)[tid] = out2 / l2;
-    reinterpret_cast<device float2*>(output + 3 * head_dim)[tid] = out3 / l3;
-    reinterpret_cast<device float2*>(output + 4 * head_dim)[tid] = out4 / l4;
-    reinterpret_cast<device float2*>(output + 5 * head_dim)[tid] = out5 / l5;
-    reinterpret_cast<device float2*>(output + 6 * head_dim)[tid] = out6 / l6;
-    reinterpret_cast<device float2*>(output + 7 * head_dim)[tid] = out7 / l7;
+
+    // Optimized output normalization with better numerical stability
```

Comment:
Explain why it is expected to have better numerical stability. Generally, division is more accurate than multiplication by the inverse.